app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -87,6 +87,9 @@ def create
       def update
         datacenter = change_datacenter_to_uuid(compute_resource_params[:datacenter])
         update_parameters = datacenter.present? ? compute_resource_params.merge(:datacenter => datacenter) : compute_resource_params
+        if @compute_resource.provider == 'vmware'
+          ComputeAttribute.where(compute_resource_id: @compute_resource.id).each { |ca| ca.vm_attrs['path'].gsub(@compute_resource.datacenter, datacenter) }
```

Comment:
This will not work, it will not save the changes to CAs.
This might, if we also add the `autosave: true` to the has_many :compute_attributes relation on ComputeResource
```suggestion
          @compute_resource.compute_attributes.each { |ca| ca.vm_attrs['path'].gsub(@compute_resource.datacenter, datacenter) }
```
Issues with this approach tho:
1. `compute_resource.datacenter` is a `uuid` not a name, we would need to get name of current datacenter I believe.
2.  one more caveat, we don't really know if the new path is correct :(
If you're renaming datacenter then yes, but are there situations where user just decides to change datacenter? (probably not, as it would render all the resources obsolete

Reply:
I checked in the case of vmware the compute_resource.datacenter returns the datacenter name and not the uuid.